### DISCUSSION TITLE
Support app passwords

### DISCUSSION THREAD

**Discussion by ibotty** [keycloak/keycloak#12503] [Ideas]
Some applications or even whole protocols don't (yet) support OIDC or SAML for authentication.  It would be great to have a way to store and manage "app passwords", that is single-purpose passwords, tied to a user, a service name and a client-identifier chosen by the user.

My usecase is IMAP and SMTP authentication for legacy clients (like thunderbird) which don't support OAuth2 (yet).

I'd propose an authenticated http endpoint that gets the user's username, password and service's name and returns either OK.  It might also return a valid token for that user, but I don't need that for my usecase.  That would enable legacy proxies in front of the OAuth2 enabled service.

**Comment by ibotty** [comment]
BTW:  I started a similar discussion in zitadel/zitadel#3707

**Reply by syrosh** [reply]
If you still need this, check out: https://github.com/radicallyopensecurity/keycloak-app-passwords
